automation/system_test/prepare.py

```diff
@@ -354,6 +376,34 @@ def _download_provctl(self):
         self._run_command("chmod", args=["+x", provctl])
         return provctl
 
+    def _wait_until_finished(
+        self,
+        command: str,
+        command_name_to_wait_for: str,
```

Comment:
```suggestion
        command_name: str,
```
I don't think you need such an explicit name, as it is inferred from the method name that you are waiting for *this* command